text/0000-stabilize-no_std.md

```diff
@@ -0,0 +1,267 @@
+(and probably should not) to define these items.
+
+#### `panic_fmt`
+
+This lang item is the definition of how to panic in Rust. The standard library
+defines this by throwing an exception (in a platform-specific manner), but users
+of libcore often want to define their own meaning of panicking. The signature of
+this function will be:
+
+```rust
+#[panic_fmt]
+pub extern fn panic_fmt(msg: &core::fmt::Arguments) -> !;
+```
+
+This differs with the `panic_fmt` function today in that the file and line
+number arguments are omitted. The libcore library will continue to provide
+file/line number information in panics (as it does today) by assembling a new
+`core::fmt::Arguments` value which uses the old one and appends the file/line
+information.
+
+This signature also differs from today's implementation by taking a `&Arguments`
+instead of taking it by value, and the purpose of this is to ensure that the
+function has a clearly defined ABI on all platforms in case that is required.
+
+#### `eh_personality`
+
+The compiler will continue to compile libcore with landing pads (e.g. cleanup to
+run on panics), and a "personality function" is required by LLVM to be available
+to call for each landing pad. In the current implementation of panicking, a
+personality function is typically just calling a standard personality function
+in libgcc (or in MSVC's CRT), but the purpose is to indicate whether an
+exception should be caught or whether cleanup should be run for this particular
+landing pad and exception combination.
+
+The exact signature of this function is quite platform-specific, but many users
+of libcore will never actually call this function as exceptions will not be
+thrown (many will likely compile with `-Z no-landing-pads` anyway). As a result
+the signature of this lang item will not be defined, but instead it will simply
+be required to be defined (as libcore will reference the symbol name
+regardless).
+
+```rust
+#[eh_personality]
+pub extern fn eh_personality(...) -> ...;
+```
+
+The compiler will not check the signature of this function, but it will assign
+it a known symbol so libcore can be successfully linked.
+
+#### `stack_exhausted`
+
+The current implementation of stack overflow in the compiler is to use LLVM's
+segmented stack support, inserting a prologue to every function in an object
+file to detect when a stack overflow occurred. When a stack overflow is
+detected, LLVM emits code that will call the symbol `__morestack`, which the
+Rust distribution provides an implementation of. Our implementation, however,
+then in turn calls a this `stack_exhausted` language item to define the
+implementation of what happens on stack overflow.
+
+The compiler therefore needs to ensure that this lang item is present in order
+for libcore to be correctly linked, so the lang item will have the following
+signature:
+
+```rust
+#[stack_exhausted]
+pub extern fn stack_exhausted() -> !;
```

Comment:
We can presumably generalise the signature in future if we find use-cases for other ones.